nixos/modules/system/boot/loader/raspberrypi/raspberrypi-builder.nix

```
@@ -3,6 +3,9 @@
 pkgs.substituteAll {
   src = ./raspberrypi-builder.sh;
   isExecutable = true;
+  postInstall = "shellcheck $out";
+  nativeBuildInputs = [pkgs.shellcheck];
```

I need to verify, but I recall that `nativeBuildInputs` and splicing won't work during cross-compilation when done like here into a NixOS module's expression.

The solution is to use `pkgs.buildPackages` rather than `pkgs`.

```suggestion
  nativeBuildInputs = [pkgs.buildPackages.shellcheck];
```